File: app/src/main/java/com/example/notes/ListAdapterV2.java
package com.example.notes;

import static com.example.notes.MainActivity.notes_text_color;
import static com.example.notes.MainActivity.notes_text_color_first;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

public class ListAdapterV2 extends RecyclerView.Adapter<ListAdapterV2.ViewHolder> {

    CardsSource dateSource;
    private OnItemClickListener itemClickListener;
    private final Fragment fragment;
    private int menuPosition;

    public ListAdapterV2(CardsSource dateSource, Fragment fragment) {
        this.dateSource = dateSource;
        this.fragment = fragment;
    }

    public int getMenuPosition() {
        return menuPosition;
    }

    private void registerContextMenu(View itemView) {
        if (fragment != null) {
            fragment.registerForContextMenu(itemView);
        }
    }

    public void setItemClickListener(OnItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.setData(dateSource.getCardData(position));
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View v = LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.item_v2, viewGroup, false);
        return new ViewHolder(v);
    }

    @Override
    public int getItemCount() {
        return dateSource.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView title;
        // private final TextView description;
        private final AppCompatImageView image;
        private final CheckBox like;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.title);
            // description = itemView.findViewById(R.id.description);
            image = itemView.findViewById(R.id.imageView);
            like = itemView.findViewById(R.id.like);
            registerContextMenu(itemView);
            image.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {

                    int position = getAdapterPosition();
                    if (itemClickListener != null)
                        itemClickListener.onItemClick(view, position);
                }
            });
            image.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    menuPosition = getLayoutPosition();
                    itemView.showContextMenu(15, 15);

                    return true;
                }
            });
        }

        public void setData(CardData cardData) {
            title.setText(cardData.getTitle());
            title.setId(cardData.getId());
            if (notes_text_color == 0) notes_text_color = notes_text_color_first;
            title.setTextColor(notes_text_color);
            //    description.setText(cardData.getDescription());
            like.setChecked(cardData.isLike());
            image.setImageResource(cardData.getPicture());

        }
    }

}

File: app/src/main/java/com/example/notes/ListAdapterV2.kt
package com.example.notes

import com.example.notes.CardsSource.setNewData
import com.example.notes.CardsSource.getCardData
import com.example.notes.CardsSource.size
import com.example.notes.CardData.title
import com.example.notes.CardData.id
import com.example.notes.CardData.isLike
import com.example.notes.CardData.picture
import com.example.notes.CardsSource
import androidx.recyclerview.widget.RecyclerView
import com.example.notes.CardData
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.notes.R
import android.widget.TextView
import androidx.appcompat.widget.AppCompatImageView
import android.widget.CheckBox
import android.view.View.OnLongClickListener
import androidx.fragment.app.Fragment
import com.example.notes.MainActivity
import java.util.ArrayList

class ListAdapterV2(var dateSource: CardsSource, private val fragment: Fragment?) :
    RecyclerView.Adapter<ListAdapterV2.ViewHolder>() {
    private var itemClickListener: OnItemClickListener? = null
    var menuPosition = 0

    fun setNewData(dataSource: ArrayList<CardData?>?) {
        dateSource.setNewData(dataSource)
        notifyDataSetChanged()
    }

    private fun registerContextMenu(itemView: View) {
        fragment?.registerForContextMenu(itemView)
    }

    fun setItemClickListener(itemClickListener: OnItemClickListener?) {
        this.itemClickListener = itemClickListener
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.setData(dateSource.getCardData(position))
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(viewGroup.context)
            .inflate(R.layout.item_v2, viewGroup, false)
        return ViewHolder(v)
    }

    override fun getItemCount(): Int {
        return dateSource.size()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val title: TextView

        // private final TextView description;
        private val image: AppCompatImageView
        private val like: CheckBox

        init {
            title = itemView.findViewById(R.id.title)
            // description = itemView.findViewById(R.id.description);
            image = itemView.findViewById(R.id.imageView)
            like = itemView.findViewById(R.id.like)
            registerContextMenu(itemView)
            image.setOnClickListener { view ->
                val position = adapterPosition
                if (itemClickListener != null) itemClickListener!!.onItemClick(view, position)
            }
            image.setOnLongClickListener {
                menuPosition = layoutPosition
                itemView.showContextMenu(15f, 15f)
                true
            }
        }

        fun setData(cardData: CardData?) {
            title.text = cardData!!.title
            title.id = cardData.id
            if (MainActivity.notes_text_color == 0) MainActivity.notes_text_color =
                MainActivity.notes_text_color_first
            title.setTextColor(MainActivity.notes_text_color)
            //    description.setText(cardData.getDescription());
            like.isChecked = cardData.isLike
            image.setImageResource(cardData.picture)
        }
    }
}